kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -1977,6 +1977,21 @@ object KyuubiConf {
       .stringConf
       .createOptional
 
+  val GROUP_PROVIDER: ConfigEntry[String] =
+    buildConf("kyuubi.session.group.provider")
+      .doc("A group provider plugin for Kyuubi Server. This plugin can provide primary group " +
+        "and groups information for different user or session configs. This config value " +
+        "should be a class which is a child of 'org.apache.kyuubi.plugin.GroupProvider' which " +
+        "has zero-arg constructor. Kyuubi provides the following built-in implementations: " +
+        "<li>hadoop: delegate the user group mapping to hadoop UserGroupInformation.</li>")
+      .version("1.7.0")
+      .stringConf
+      .transform {
+        case "hadoop" => "org.apache.kyuubi.session.HadoopGroupProvider"
```

Comment:
give the built-in a shortcut

Reply:
Whenever a built -in implementation is added, you need to change the code here and re -generate the document. It seems that it is not very friendly to users. Can't you define it like session_conf_advisor?

Reply:
We should provide the built-in implementation list in the description, and provide a shortcut like "hadoop", "ldap", "unix" for convenient, but allow users to set the full class name for third party implementation.
I'm planning to change to the `SESSION_CONF_ADVISOR`'s documentation to align w/ this way, because it does not clear which built-in implementations are available.